onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc

```diff
@@ -2502,32 +2484,25 @@ Status GatherOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const
       indices_data_type != ONNX_NAMESPACE::TensorProto_DataType_INT32) {
     // Add indices operand into nnapi
     const auto& indices_tensor = *initializers.at(input2);
-    std::vector<uint8_t> unpacked_tensor;
-    ORT_RETURN_IF_ERROR(onnxruntime::utils::UnpackInitializerData(indices_tensor, unpacked_tensor));
+    Initializer unpacked_tensor(indices_tensor);
 
     const auto data_type = indices_tensor.data_type();
     const auto indices_shape = indices_tensor.dims();
-    uint32_t size = 1;
     Shape indices_dimen;
     indices_dimen.reserve(indices_tensor.dims_size());
     for (auto i = 0; i < indices_tensor.dims_size(); i++) {
-      size *= SafeInt<uint32_t>(indices_shape[i]);
       indices_dimen.push_back(static_cast<uint32_t>(indices_shape[i]));
     }
 
-    std::vector<int32_t> indices(size);
+    std::vector<int32_t> indices(unpacked_tensor.size());
     // see https://gist.github.com/shafik/848ae25ee209f698763cffee272a58f8#type-punning-arrays for the usage of memcpy here
     if (data_type == ONNX_NAMESPACE::TensorProto_DataType_INT64) {
-      for (uint32_t i = 0; i < size; i++) {
-        int64_t index_i64;
-        memcpy(&index_i64, unpacked_tensor.data() + i * sizeof(int64_t), sizeof(int64_t));
-        indices[i] = SafeInt<int32_t>(index_i64);
+      for (uint32_t i = 0; i < unpacked_tensor.size(); i++) {
+        indices[i] = SafeInt<int32_t>(unpacked_tensor.DataAsSpan<int64_t>()[i]);
       }
```

Comment:
This change means the comment on line 2498 no longer applies. @edgchen1 would it be better/faster to use memcpy here? Or maybe std::transform?

Reply:
yeah we can delete that comment. I think std::transform is a good option. memcpy seemed more correct before but maybe I was being overly pedantic.